Enum members with commas cause parser crash (nil pointer dereference)

## Description

Defining enum members separated by commas causes a nil pointer dereference in the parser, crashing the program.

## Severity

**Critical** - This is a crash bug that terminates the program unexpectedly.

## Reproduction

```ez
const Status enum {
    PENDING,
    ACTIVE,
    INACTIVE
}
```

## Error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x...]
```

## Expected Behavior

Either:
1. Commas should be accepted as valid separators between enum members, OR
2. A clear, helpful error message should be shown indicating that commas are not allowed

## Workaround

Use newlines without commas between enum members:

```ez
const Status enum {
    PENDING
    ACTIVE
    INACTIVE
}
```

## Additional Context

Found during edge case testing on the `feat/module-system` branch. Test file: `test/edge_case_structs_enums.ez`

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -590,13 +590,21 @@ func (p *Parser) parseVarableDeclarationOrStruct() Statement {
 		if p.peekTokenMatches(STRUCT) {
 			// This is a struct declaration
 			// currentToken is now the struct name, which is what parseStructDeclaration expects
-			return p.parseStructDeclaration()
+			result := p.parseStructDeclaration()
+			if result == nil {
+				return nil // Return untyped nil to avoid typed nil interface issue
+			}
+			return result
 		}
 
 		if p.peekTokenMatches(ENUM) {
 			// This is an enum declaration
 			// currentToken is now the enum name, which is what parseEnumDeclaration expects
-			return p.parseEnumDeclaration()
+			result := p.parseEnumDeclaration()
+			if result == nil {
+				return nil // Return untyped nil to avoid typed nil interface issue
+			}
+			return result
 		}
 
 		// Not a struct, restore and parse as variable
@@ -1540,10 +1548,39 @@ func (p *Parser) parseEnumDeclaration() *EnumDeclaration {
 	stmt.Name = &Label{Token: p.currentToken, Value: p.currentToken.Literal}
 
 	p.nextToken() // move past name to 'enum'
-	p.nextToken() // move past 'enum' to '{'
+	p.nextToken() // move past 'enum' to ':' or '{'
+
+	// Check for type annotation: enum : type { ... }
+	if p.currentTokenMatches(COLON) {
+		p.nextToken() // move past ':' to type name
+
+		if !p.currentTokenMatches(IDENT) {
+			msg := fmt.Sprintf("expected type name after ':', got %s", p.currentToken.Type)
+			p.addEZError(errors.E2024, msg, p.currentToken)
+			return nil
+		}
+
+		typeName := p.currentToken.Literal
+
+		// Validate that type is a primitive (int, float, or string)
+		if typeName != "int" && typeName != "float" && typeName != "string" {
+			msg := fmt.Sprintf("enum type must be a primitive type (int, float, or string), got '%s'", typeName)
+			p.addEZError(errors.E2026, msg, p.currentToken)
+			return nil
+		}
+
+		// Initialize attributes with the specified type
+		stmt.Attributes = &EnumAttributes{
+			TypeName:  typeName,
+			Skip:      false,
+			Increment: nil,
+		}
+
+		p.nextToken() // move past type name to '{'
+	}
 
 	if !p.currentTokenMatches(LBRACE) {
-		msg := "expected '{' after enum name"
+		msg := "expected '{' after enum declaration"
 		p.addEZError(errors.E2002, msg, p.currentToken)
 		return nil
 	}

